immutablecollections/_immutabledict.py

```diff
@@ -32,7 +34,7 @@
 
 
 def immutabledict(
-    iterable: Optional[AllowableSourceType] = None
+    iterable: Optional[AllowableSourceType] = None, forbid_duplicate_keys: bool = False
```

Comment:
I think `forbid_duplicate_keys` needs to be a keyword-only argument by our coding standards as it's boolean. (I don't know where those standards are; @rgabbard should be able to point you there.)

Reply:
TIL how keyword-only arguments work: https://www.python.org/dev/peps/pep-3102/

Yes, we can wait for Ryan, since I couldn't find anything in the VistaNLP coding standards document. But to be clear what we're talking about, here are some examples from `vistautils.parameters`:
```python
    def creatable_empty_directory(self, param: str, *, delete=False) -> Path:
    def optional_evaluate(
        self,
        name: str,
        expected_type: Type[ParamType],
        *,
        namespace_param_name: str = "value",
        special_values: Mapping[str, str] = ImmutableDict.empty(),
    ) -> Optional[ParamType]:
    def path_list_from_file(self, param: str, *, log_name=None) -> Sequence[Path]:
```
I didn't see anything regarding consistent usage, but since Constantine brought it up, let's make it keyword-only.